application/models/TemplateConfiguration.php

```diff
@@ -1354,4 +1380,188 @@ public function setOptionKeysToInherit(){
+     * @return LimeSurvey\Datavalueobjects\ThemeFileCategory[]
+     */
+    public function getFileCategories()
+    {
+        // We need to determine the paths first. They may already be set, but if they're not, we need to get them from the template.
+        // Note that the template cannot be accessed as relation until the model is saved.
+        $template = !empty($this->template) ? $this->template : Template::model()->findByAttributes(['name' => $this->template_name]);
+        if (empty($this->path)) {
+            $isStandard = Template::isStandardTemplate($this->template_name);
+            $path = $isStandard
+                ? Yii::app()->getConfig("standardthemerootdir") . DIRECTORY_SEPARATOR . $template->folder . DIRECTORY_SEPARATOR
+                : Yii::app()->getConfig("userthemerootdir") . DIRECTORY_SEPARATOR . $template->folder . DIRECTORY_SEPARATOR;
+        } else {
+            $path = $this->path;
+        }
+        $generalFilesPath = !empty($this->generalFilesPath) ? $this->generalFilesPath : Yii::app()->getConfig("userthemerootdir") . DIRECTORY_SEPARATOR . 'generalfiles' . DIRECTORY_SEPARATOR;
+
+        /** @var LimeSurvey\Datavalueobjects\ThemeFileCategory[] */
+        $categoryList = [];
+        $categoryList[] = new ThemeFileCategory('generalfiles', gT("Global"), $generalFilesPath, 'image::generalfiles::');
+        $categoryList[] = new ThemeFileCategory('theme', gT("Theme"), $path, 'image::theme::');
+        if (!empty($this->sid)) {
+            $categoryList[] = new ThemeFileCategory('survey', gT("Survey"), Yii::app()->getConfig('uploaddir') . '/surveys/' . $this->sid . '/images/', 'image::survey::');
+        }
+        return $categoryList;
+    }
+
+    /**
+     * Returns the virtual path of $path if it's valid.
+     *
+     * @param string $path  the path to check. Can be a "virtual" path (eg. 'image::theme::logo.png'), or a normal path.
+     *
+     * @return string|null the virtual path if it's valid, of null if it's not.
+     */
+    public function getVirtualThemeFilePath($path)
+    {
+        /** @var LimeSurvey\Datavalueobjects\ThemeFileInfo|null */
+        $fileInfo = $this->getThemeFileInfo($path);
+
+        if (empty($fileInfo)) {
+            return null;
+        }
+
+        return $fileInfo->virtualPath;
+    }
+
+    /**
+     * Returns the real absolute path of $path if it's valid.
+     *
+     * @param string $path  the path to check. Can be a "virtual" path (eg. 'image::theme::logo.png'), or a normal path.
+     *
+     * @return string|null the real absolute path if it's valid, of null if it's not.
+     */
+    public function getRealThemeFilePath($path)
+    {
+        /** @var LimeSurvey\Datavalueobjects\ThemeFileInfo|null */
+        $fileInfo = $this->getThemeFileInfo($path);
+
+        if (empty($fileInfo)) {
+            return null;
+        }
+
+        return $fileInfo->realPath;
+    }
+
+    /**
+     * Returns the theme file information if the path is valid.
```

Comment:
Validates $path and returns its file info
Rules:
- valid virtual path 
- real path for an available category